### PR TITLE
[ready2merge][orc8r][helm] Promote script for helm charts

### PR DESCRIPTION
**Pull request by quentinDERORY**
## Summary

Script is copying specified artifacts from one repo to another
Aimed to be used to promote helm charts artifacts after tests passed

## Test Plan

Tested to promote helm-test artifacts to helm-prod successfully.
(Did not kept artifacts in helm-prod as not entirely tested)

## Additional Information

Needed Environment variables:
HELM_CHART_ARTIFACT_URL: https://artifactory.magmacore.org:443/artifactory/
HELM_CHART_MUSEUM_ORIGIN_REPO: helm-test
HELM_CHART_MUSEUM_DEST_REPO: helm-prod
HELM_CHART_MUSEUM_USERNAME
HELM_CHART_MUSEUM_TOKEN

Usage : $MAGMA_ROOT/orc8r/tools/helm/promote.sh orc8r-1.5.tgz fbinternal_orc8r-0.2.1.tgz

### CODE REVIEW

File: orc8r/tools/helm/promote.sh [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  echo "$1"
+  exit 1
+}
+
+# Check if the required args and env-vars present
+[ $# -eq 0 ] && usage
+
+if [[ -z $HELM_CHART_ARTIFACTORY_URL ]]; then
+  HELM_CHART_ARTIFACTORY_URL="https://artifactory.magmacore.org:443/artifactory"
+fi
```
Comment (hcgatewood) [inline orc8r/tools/helm/promote.sh]:
```suggestion
HELM_CHART_ARTIFACTORY_URL=${HELM_CHART_ARTIFACTORY_URL:-https://artifactory.magmacore.org:443/artifactory}
```

Comment (hcgatewood) [reply orc8r/tools/helm/promote.sh]: And same pattern for the below default values

File: orc8r/tools/helm/promote.sh [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+set -e -o pipefail
+
+# Form API URL
+HELM_CHART_MUSEUM_API_URL="$HELM_CHART_ARTIFACTORY_URL/api"
+
+# iterate through artifacts to promote
+for artifact in "$@"
+do
+    set +e +o pipefail
```
Comment (hcgatewood) [inline orc8r/tools/helm/promote.sh]: I think there might be a way to get curl to *not* exit with error on bad request. That would be preferable to this verbose option toggling. Lmk if you're able to get that to work!

File: orc8r/tools/helm/promote.sh [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      promote_artifact "$artifact"
+    elif [ "$RESPONSE_CODE_ARTIFACT" == "404" ]; then
+      exitmsg "The artifact $artifact was not found in repository $HELM_CHART_MUSEUM_ORIGIN_REPO"
+    else
+      exitmsg "There was an error retrieving $artifact from repository $HELM_CHART_MUSEUM_ORIGIN_REPO"
+    fi
+done
+
+printf '\n'
+echo "Promoted orc8r chart artifacts $* from $HELM_CHART_MUSEUM_ORIGIN_REPO to $HELM_CHART_MUSEUM_DEST_REPO successfully."
```
Comment (hcgatewood) [inline orc8r/tools/helm/promote.sh]:
```suggestion
echo "Promoted Orc8r chart artifacts $* from $HELM_CHART_MUSEUM_ORIGIN_REPO to $HELM_CHART_MUSEUM_DEST_REPO successfully."
```

File: orc8r/tools/helm/promote.sh [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# promote.sh copies specified artifacts from one repo to another
```
Comment (hcgatewood) [inline orc8r/tools/helm/promote.sh]:
```suggestion
# promote.sh copies specified artifacts from one repo to another.
```

File: orc8r/tools/helm/promote.sh (hunk 2 above)
Comment (hcgatewood) [reply orc8r/tools/helm/promote.sh]: If not, encapsulate every usage of the option toggle within a function, for added readability. E.g. "set +e ..., get_artifact, set -e ..." encapsulated in a single function

**Review by hcgatewood — APPROVED** [review]
Looks good, couple style/conciseness comments

File: orc8r/tools/helm/promote.sh (hunk 2 above)
Comment (quentinDERORY) [reply orc8r/tools/helm/promote.sh]: got it working with `|| :` !

**Pull request merged**